src/amo/components/AddonCompatibilityError/index.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable react/no-danger */
+import { oneLine } from 'common-tags';
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import {
+  INCOMPATIBLE_FIREFOX_FOR_IOS,
+  INCOMPATIBLE_NOT_FIREFOX,
+  INCOMPATIBLE_OVER_MAX_VERSION,
+  INCOMPATIBLE_UNDER_MIN_VERSION,
+} from 'core/constants';
+import translate from 'core/i18n/translate';
+import _log from 'core/logger';
+import { sanitizeHTML } from 'core/utils';
+
+import './style.scss';
+
+
+export class AddonCompatibilityErrorBase extends React.Component {
+  static propTypes = {
+    i18n: PropTypes.object.isRequired,
+    lang: PropTypes.string.isRequired,
+    log: PropTypes.object,
+    maxVersion: PropTypes.string.isRequired,
+    minVersion: PropTypes.string.isRequired,
+    reason: PropTypes.string.isRequired,
+    userAgentInfo: PropTypes.object.isRequired,
+  }
+
+  static defaultProps = {
+    log: _log,
+  }
+
+  render() {
+    const {
+      i18n,
+      lang,
+      log,
+      maxVersion,
+      minVersion,
+      reason,
+      userAgentInfo,
```

Comment:
On the client side, it's possible for any component to render for a split second without any redux data. This could happen during navigation from one page to another but it won't ever happen server side (the redux data will always be pre-loaded).